File: aulas-etecvav/3bim/Programação-e-algoritimo/python-conversion/cpp/ex08_numero_primo.cpp
/*
    Autor: Kael Elodie Whitmore
    Data: Junho/2026
    Descrição: Verifica se um número inteiro positivo é primo.
*/

#include <iostream>
using namespace std;

int main()
{
    int numero;
    int divisor = 1;
    int quantidadeDivisores = 0;

    cout << "Digite um numero inteiro positivo: ";
    cin >> numero;

    if (numero > 0)
    {
        while (divisor <= numero)
        {
            if (numero % divisor == 0)
            {
                quantidadeDivisores++;
            }

            divisor++;
        }

        if (quantidadeDivisores == 2)
        {
            cout << "O numero e primo." << endl;
        }
        else
        {
            cout << "O numero nao e primo." << endl;
        }
    }
    else
    {
        cout << "Numero invalido." << endl;
    }

    return 0;
}

File: aulas-etecvav/3bim/Programação-e-algoritimo/python-conversion/python/ex08_numero_primo.py
# Autor: Matheus F. Mation
# Data: Agosto/2026 
# Descrição: Verifica se um número inteiro positivo é primo.

divisor = 1
quantidadeDivisores = 0
numero = int(input("Digite um numero inteiro positivo: "))
if numero > 0:
    while divisor <= numero:
        if numero % divisor == 0:
            quantidadeDivisores += 1
        divisor += 1
    if quantidadeDivisores == 2:
        print("O numero e primo.")
    else:
        print("O numero nao e primo.")
else:
    print("numero invalido")
